spring-cloud-dataflow-tasklauncher/spring-cloud-dataflow-tasklauncher-function/src/main/java/org/springframework/cloud/dataflow/tasklauncher/TaskLauncherFunction.java

```diff
@@ -43,34 +43,30 @@
  *
  * @author David Turanski
  **/
-public class TaskLauncherFunction implements Function<LaunchRequest, Optional<LaunchResponse>>, InitializingBean {
-	private static final Log log = LogFactory.getLog(TaskLauncherFunction.class);
+public class TaskLauncherFunction implements Consumer<LaunchRequest>, InitializingBean {
 
-	static final String TASK_PLATFORM_NAME = "spring.cloud.dataflow.task.platformName";
+	private static final Logger log = LoggerFactory.getLogger(TaskLauncherFunction.class);
 
-	private final TaskOperations taskOperations;
+	public static final String TASK_PLATFORM_NAME = "spring.cloud.dataflow.task.platformName";
 
+	private final TaskOperations taskOperations;
 	private String platformName = "default";
 
 	public TaskLauncherFunction(TaskOperations taskOperations) {
 		Assert.notNull(taskOperations, "`taskOperations` cannot be null.");
 		this.taskOperations = taskOperations;
 	}
 
-	/**
-	 *
-	 * @param launchRequest the task launch request for the Data Flow server.
-	 * @return an {@code Optional<Long>} containing the task Id if the request is accepted or
-	 * empty otherwise.
-	 */
 	@Override
-	public Optional<LaunchResponse> apply(LaunchRequest launchRequest) {
+	public void accept(LaunchRequest request) {
 		if (platformIsAcceptingNewTasks()) {
-			return Optional.of(launchTask(launchRequest));
+			log.debug("task-launcher-function:request:{}", request);
+			LaunchResponse response = launchTask(request);
+			log.debug("task-launcher-function:response:{}", response);
+		} else {
+			log.warn("Platform is at capacity. Did not submit task launch request for task {}.", request.getTaskName());
+			throw new CannotHandleRequestException();
```

Comment:
~Does this cause the retry code to kick in? Is that the significance of this?~
I see https://github.com/spring-cloud/spring-cloud-dataflow/pull/5664/files#diff-0ef2bb2c73eccaefa2bbc942197dd1fbb45dfcf36b1c19735a7015fa03e0a70eR70